### PR TITLE
Array shuffle

### PR DESCRIPTION
**Pull request by mediacloner**
New exercise done! With testing. 

Please, remove  visual code folder. I add exception in git ignore but don't work properly.

### CODE REVIEW

File: course/frontend/array-shuffle/scripts/main.js [hunk 1]
```diff
@@ -1 +1,16 @@
-// TODO
\ No newline at end of file
+describe("Array", function() {
```
[the remark this answers is not in the document]
Comment (mediacloner) [reply course/frontend/array-shuffle/scripts/main.js]: Rename and modify in specrunner.html

File: .vscode/launch.json [hunk 2]
```diff
@@ -0,0 +1,14 @@
+{
```
[the remark this answers is not in the document]
Comment (mediacloner) [reply .vscode/launch.json]: Done!

File: course/frontend/array-shuffle/SpecRunner.html [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+
+  <script src="vendor/jasmine/2.9.0/jasmine.js"></script>
+  <script src="vendor/jasmine/2.9.0/jasmine-html.js"></script>
+  <script src="vendor/jasmine/2.9.0/boot.js"></script>
+
+  <!-- include source files here... -->
+  <script src="scripts/array-shuffle-polyfill.js"></script>
+
+  <!-- include spec files here... -->
+  <script src="scripts/array-shuffle-polyfill.js"></script>
```
[the remark this answers is not in the document]
Comment (mediacloner) [reply course/frontend/array-shuffle/SpecRunner.html]: Done!

File: course/frontend/array-shuffle/scripts/main.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (mediacloner) [reply course/frontend/array-shuffle/scripts/main.js]: Done!

File: course/frontend/hangman/scripts/Hangman.js [hunk 4]
```diff
@@ -1,3 +1,4 @@
+
```
[the remark this answers is not in the document]
Comment (mediacloner) [reply course/frontend/hangman/scripts/Hangman.js]: Done!

File: course/frontend/hangman/tests/HangmanSpec.js [hunk 5]
```diff
@@ -1,31 +1,129 @@
...
-        expect(game.try('d')).toBe('6 _ _ _ _ _');
-        expect(game.try('e')).toBe('6 _ E _ _ _');
-        expect(game.try('l')).toBe('6 _ E L L _');
-        expect(game.try('bello')).toBe('Sorry, you have not guessed... the correct word is HELLO.');
-    }
-
-    it(hangmanResultsSorryWhenFailingAssertionOfWordOnThe7thAttempt.name, hangmanResultsSorryWhenFailingAssertionOfWordOnThe7thAttempt);
-});
\ No newline at end of file
+describe("Hangman", function() {
```
[the remark this answers is not in the document]
Comment (mediacloner) [reply course/frontend/hangman/tests/HangmanSpec.js]: Done!

File: course/frontend/prototyping/.vscode/launch.json [hunk 6]
```diff
@@ -0,0 +1,14 @@
+{
```
[the remark this answers is not in the document]
Comment (mediacloner) [reply course/frontend/prototyping/.vscode/launch.json]: Done!

File: course/frontend/prototyping/README.md [hunk 7]
```diff
@@ -0,0 +1,67 @@
+# Challenge Prototype & Inheritance [2h]
```
[the remark this answers is not in the document]
Comment (mediacloner) [reply course/frontend/prototyping/README.md]: Done!

**Comment by mediacloner** [comment]
I merge some files because there was conflict. Warning: I select my code and discard Didac's code. Probably it's not the best option.

File: course/frontend/array-shuffle/SpecRunner.html [hunk 8]
```diff
@@ -1,24 +1,22 @@
 <!DOCTYPE html>
```
Comment (manuelbarzi) [inline course/frontend/array-shuffle/SpecRunner.html]: UPDATE this branch with upstream/develop, RESET it if necessary, and then apply your changes without affecting the existing code ,)

Comment (mediacloner) [reply course/frontend/array-shuffle/SpecRunner.html]: I think so It's done.

File: course/frontend/array-shuffle/scripts/array-shuffle-polyfill.js [hunk 9]
```diff
@@ -1,32 +1,22 @@
...
-	return newArray;
-}
-
-// WARN! this is required to avoid 'shuffle' to be present in an array for-in loop!!!
-Object.defineProperty(Array.prototype, 'shuffle', {
-	enumerable: false
-});
\ No newline at end of file
+
+Array.prototype.shuffle = function() {
```
Comment (manuelbarzi) [inline course/frontend/array-shuffle/scripts/array-shuffle-polyfill.js]: WELL done! but please, do not delete the solution provided before. KEEP both solutions, and comment with a low-dash the one you want to "comment" (e.g. Array.prototype._shuffle).

Comment (mediacloner) [reply course/frontend/array-shuffle/scripts/array-shuffle-polyfill.js]: You all right! It's better better modify than destroy. I include the lost lines.

**Pull request merged**